### PR TITLE
Implement Page.GoBackAsync and Page.GoForwardAsync

### LINKED ISSUE #214: Implement Page.goBack
https://github.com/GoogleChrome/puppeteer/blob/v1.0.0/test/test.js#L1231

### PR DESCRIPTION
**Pull request by kblok**
Closes #214

### CODE REVIEW

**Comment by kblok** [comment]
Ready to review @Meir017

File: lib/PuppeteerSharp/Page.cs [hunk 1]
```diff
@@ -971,10 +971,54 @@ public Task<ElementHandle> WaitForSelectorAsync(string selector, WaitForSelector
...
         }
 
+        /// <summary>
+        /// Navigate to the previous page in history.
+        /// </summary>
+        /// <returns>Task which which resolves to the main resource response. In case of multiple redirects, 
+        /// the navigation will resolve with the response of the last redirect. If can not go back, resolves to null.</returns>
+        /// <param name="options">Navigation parameters.</param>
+        public Task<Response> GoBackAsync(NavigationOptions options = null)
+        {
```
Comment (Meir017) [inline lib/PuppeteerSharp/Page.cs]: Arrow

File: lib/PuppeteerSharp/Page.cs [hunk 2]
```diff
@@ -971,10 +971,54 @@ public Task<ElementHandle> WaitForSelectorAsync(string selector, WaitForSelector
...
+        }
+
+        /// <summary>
+        /// Navigate to the next page in history.
+        /// </summary>
+        /// <returns>Task which which resolves to the main resource response. In case of multiple redirects, 
+        /// the navigation will resolve with the response of the last redirect. If can not go forward, resolves to null.</returns>
+        /// <param name="options">Navigation parameters.</param>
+        public Task<Response> GoForwardAsync(NavigationOptions options = null)
+        {
```
Comment (Meir017) [inline lib/PuppeteerSharp/Page.cs]: Arrow

File: lib/PuppeteerSharp/Page.cs [hunk 3]
```diff
@@ -971,10 +971,54 @@ public Task<ElementHandle> WaitForSelectorAsync(string selector, WaitForSelector
...
+            if (history.Entries.Count <= history.CurrentIndex + delta)
+            {
+                return null;
+            }
+            var entry = history.Entries[history.CurrentIndex + delta];
+            var waitTask = WaitForNavigationAsync(options);
+
+            await Task.WhenAll(
+                waitTask,
+                Client.SendAsync<ResponseData>("Page.navigateToHistoryEntry", new
```
Comment (Meir017) [inline lib/PuppeteerSharp/Page.cs]: Do we really need the class `ResponseData`? It's never used

Comment (kblok) [reply lib/PuppeteerSharp/Page.cs]: No. It was a approach I didn't take.

**Comment by kblok** [comment]
Done @Meir017

**Pull request merged**